typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,628 @@
+                )
+            ];
+        }
+
+        const serviceContext: ServiceContext = new ServiceContext(
+            this.client,
+            serviceId,
+            serviceResponse.getIpAddr(),
+            this.enclaveDataVolMountpoint,
+            enclaveDataVolMountDirpathOnSvcContainer,
+        )
+
+        return [serviceContext, null];
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public async removeService(serviceId: ServiceID, containerStopTimeoutSeconds: number): Promise<Error> {
+
+        winston.debug("Removing service '%v'...", serviceId);
+        // NOTE: This is kinda weird - when we remove a service we can never get it back so having a container
+        //  stop timeout doesn't make much sense. It will make more sense when we can stop/start containers
+        // Independent of adding/removing them from the network
+        const args: RemoveServiceArgs = newGetRemoveServiceArgs(serviceId, containerStopTimeoutSeconds);
+        
+        // TODO TODO TODO - CALLBACK & ERROR-HANDLING (REMOVE)
+        // if _, err := networkCtx.client.RemoveService(context.Background(), args); err !== nil {
+        //     return stacktrace.Propagate(err, "An error occurred removing service '%v' from the network", serviceId)
+        // }
+
+        const promiseAsyncRemoveService: Promise<ResultAsync<any, Error>> = new Promise((resolve, _unusedReject) => {
+            this.client.removeService(args, (_unusedError: Error, response: any) => {
+                resolve(okAsync(response));
+            })
+        });
+        const promiseRemoveService: Result<any, Error> = await promiseAsyncRemoveService;
+        if (!promiseRemoveService.isOk()) {
+            return promiseRemoveService.error;
+        }
+
+        winston.debug("Successfully removed service ID %v", serviceId);
+
+        return null;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public async repartitionNetwork(
+        partitionServices: Map<PartitionID, Set<ServiceID>>,
+        partitionConnections: Map<PartitionID, Map<PartitionID, PartitionConnectionInfo>>,
+        defaultConnection: PartitionConnectionInfo): Promise<Error> {
+
+        if (partitionServices === null) {
+            return new Error("Partition services map cannot be nil");
+        }
+        if (defaultConnection === null) {
+            return new Error("Default connection cannot be nil");
+        }
+
+        // Cover for lazy/confused users
+        if (partitionConnections === null) {
+            partitionConnections = new Map();
+        }
+
+        const reqPartitionServices: Map<string, PartitionServices> = new Map();
+        for (let [partitionId, serviceIdSet] of partitionServices.entries()) {
+
+            const serviceIdStrSet: Set<string> = new Set();
+            for (let serviceId in serviceIdSet) {
+                serviceIdStrSet.add(serviceId);
```

Comment:
I noticed that you didn't need to cast `serviceId` using `String(serviceId)`. Depending on the way Typescript works (i.e. if you can use `Set<ServiceID>` equivalently with a `Set<string>`) then you may not even need to create this `serviceIdStrSet`, and can just pass this directly to `newPartitionServices(serviceIdSet)`.

The way Golang type alises work doesn't allow us to do this (e.g. `map<string, bool>` != `map<ServiceID, bool>`), but in some languages that is the case